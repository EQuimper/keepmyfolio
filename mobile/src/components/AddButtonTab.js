// @flow

import React from 'react';
import { View, StyleSheet } from 'react-native';
// ------------------------------------
// UTILS
// ------------------------------------
import { colors } from '../utils/constants';

const ADD_BUTTON_SIZE = 50;
const OUTER_SIZE = 65;

const styles = StyleSheet.create({
  addButton: {
    alignItems: 'center',
    backgroundColor: colors.primary,
    borderRadius: ADD_BUTTON_SIZE / 2,
    height: ADD_BUTTON_SIZE,
    justifyContent: 'center',
    width: ADD_BUTTON_SIZE,
  },
  root: {
    alignItems: 'center',
    borderRadius: OUTER_SIZE / 2,
    height: OUTER_SIZE,
    justifyContent: 'center',
    marginTop: -30,
    width: OUTER_SIZE,
  },
});

type Props = {
  backgroundColor: string,
  children: React.Element<*>,
};

function AddButtonTab({ children, backgroundColor }: Props) {
  return (
    <View style={[styles.root, { backgroundColor }]}>
      <View style={styles.addButton}>{children}</View>
    </View>
  );
}

export default AddButtonTab;
