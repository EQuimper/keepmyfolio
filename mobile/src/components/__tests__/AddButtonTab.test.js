// @noflow

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import AddButtonTab from '../AddButtonTab';

describe('AddButtonTab Components', () => {
  it('should renders correctly', () => {
    const component = renderer.create(<AddButtonTab />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
