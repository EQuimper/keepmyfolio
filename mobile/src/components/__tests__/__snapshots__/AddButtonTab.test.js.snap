// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButtonTab Components should renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 32.5,
        "height": 65,
        "justifyContent": "center",
        "marginTop": -30,
        "width": 65,
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00C0FF",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "width": 50,
      }
    }
  />
</View>
`;
